### PR TITLE
Undeprecate ToGrayScale transforms and functionals

### PR DESCRIPTION
**Pull request by NicolasHug**
cc @bjuncek @pmeier

### CODE REVIEW

File: torchvision/prototype/datapoints/_image.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-from typing import Any, List, Optional, Tuple, Union
+from typing import Any, List, Literal, Optional, Tuple, Union
```
Comment (pmeier) [inline torchvision/prototype/datapoints/_image.py:3]: This only works for Python >= 3.8

```suggestion
from typing import Any, List, Optional, Tuple, Union

from typing_extensions import Literal
```

**Review by pmeier — COMMENTED** [review]
Just a heads-up.

File: torchvision/prototype/datapoints/_image.py [hunk 2]
```diff
@@ -169,6 +170,12 @@ def elastic(
         )
         return Image.wrap_like(self, output)
 
+    def to_grayscale(self, num_output_channels: Literal[1, 3] = 1) -> Image:
```
Comment (NicolasHug) [inline torchvision/prototype/datapoints/_image.py:173]: Should the method be called `rgb_to_grayscale()` instead? Or just `grayscale()`?

Right now the class is called `Grayscale()`, the functionals are called `rgb_to_grayscale...()`. Keeping `to_grayscale()` for the methods seem like adding even more inconsistency, but IDK which one to go for.

We also have the `to_grayscale()` functional outlier in the v1 area which only works on PIL images (that we probably want to deprecate as discussed before).

**Comment by NicolasHug** [comment]
Should be ready for a review now @pmeier

File: torchvision/prototype/transforms/_color.py [hunk 3]
```diff
@@ -6,11 +6,47 @@
 
 from torchvision.prototype import datapoints
 from torchvision.prototype.transforms import functional as F, Transform
+from typing_extensions import Literal
```
Comment (pmeier) [inline torchvision/prototype/transforms/_color.py:9]: We should be consistent here. Sometimes you are using `: int` over `Literal[1, 3]`. I'm not going to oppose the removal the removal, but I personally would keep it given that it is going to become a language feature really soon: https://github.com/pytorch/vision/pull/7110#discussion_r1080876196

File: torchvision/prototype/transforms/functional/_color.py [hunk 4]
```diff
@@ -7,10 +9,53 @@
 
 from torchvision.utils import _log_api_usage_once
 
-from ._meta import _num_value_bits, _rgb_to_gray, convert_dtype_image_tensor
+from ._meta import _num_value_bits, convert_dtype_image_tensor
 from ._utils import is_simple_tensor
 
 
+def _rgb_to_grayscale_image_tensor(
```
Comment (pmeier) [inline torchvision/prototype/transforms/functional/_color.py:16]: Could you explain why we need this as public and private?

File: torchvision/prototype/transforms/_color.py (hunk 3 above)
Comment (NicolasHug) [reply torchvision/prototype/transforms/_color.py:9]: Literal isn't supported by torchscript.  I had it everywhere, but had to remove it later on. LMK if you prefer `int` vs `Literal` for the non-JIT cases

File: torchvision/prototype/transforms/functional/_color.py (hunk 4 above)
Comment (NicolasHug) [reply torchvision/prototype/transforms/functional/_color.py:16]: The private `_rgb_to_grayscale_image_tensor()` is also used in other functionals (see below) with `preserve_dtype=False`, which isn't a parameter we'd like to expose. This is essentially just a renaming of the previous `_rgb_to_gray()` util (which also had a `cast` param that I renamed.)

File: torchvision/prototype/transforms/_color.py (hunk 3 above)
Comment (pmeier) [reply torchvision/prototype/transforms/_color.py:9]: No, let's strife for consistency.

**Review by pmeier — APPROVED** [review]
LGTM, thanks Nicolas!

File: torchvision/prototype/datapoints/_image.py (hunk 2 above)
Comment (NicolasHug) [reply torchvision/prototype/datapoints/_image.py]: shoot, I merged before we got a chance to address that. Any thought @pmeier?

Comment (pmeier) [reply torchvision/prototype/datapoints/_image.py]: I would keep consistency with the dispatchers, i.e. naming this `rgb_to_grayscale`.

Comment (NicolasHug) [reply torchvision/prototype/datapoints/_image.py]: Following up in https://github.com/pytorch/vision/pull/7136

**Pull request merged**